src/test/scala/io/iohk/ethereum/ObjectGenerators.scala

```diff
@@ -139,6 +141,16 @@ trait ObjectGenerators {
     td <- bigIntGen
   } yield NewBlock(Block(blockHeader, BlockBody(stxs, uncles)), td)
 
+  def extraFieldsGen: Gen[HeaderExtraFields] = for {
```

Comment:
I know we have property tests for it, but maybe it is wortth while to create some specific encoding cases like in: https://github.com/input-output-hk/mantis/blob/develop/src/test/scala/io/iohk/ethereum/network/p2p/messages/ReceiptsSpec.scala#L38, it will give someone inspecting it in the future idea how encoded payload looks like.